spec/jobs/document_proofing_job_spec.rb

```diff
@@ -113,14 +113,15 @@
 
         expect(result.result).to eq(
           alert_failure_count: 0,
+          vendor: 'Acuant',
           billed: true,
           errors: {},
           face_match_results: { is_match: true, match_score: nil },
           image_metrics: {},
           processed_alerts: { failed: [], passed: [] },
           raw_alerts: [],
           raw_regions: [],
-          result: 'Passed',
+          doc_auth_result: 'Passed',
```

Comment:
This change was unexpected at first, but on closer inspection, it appears this spec is only testing the "liveness enabled" flow, where the result was a merged response:

https://github.com/18F/identity-idp/blob/005492092dce055cebf37c6f0c2b370830c59e04/app/services/doc_auth/acuant/acuant_client.rb#L77-L81

So this would have been using `DocAuth::Response#to_h`, not `DocAuth::Acuant::Response#to_h`, and would include properties previously being passed to `super`:

https://github.com/18F/identity-idp/blob/005492092dce055cebf37c6f0c2b370830c59e04/app/services/doc_auth/acuant/responses/get_results_response.rb#L12-L24

Since liveness isn't enabled in production, this now matches what was actually being logged for real-world, non-liveness outcomes, via the (now-removed) `DocAuth::Acuant::Response#to_h`:

https://github.com/18F/identity-idp/blob/005492092dce055cebf37c6f0c2b370830c59e04/app/services/doc_auth/acuant/responses/get_results_response.rb#L60-L68

Takeaways:

- This spec should include a pair of context blocks to switch `liveness_checking_enabled`  on and off and check results
- `doc_auth_result` existed previously, but maybe it should be normalized to `result` ? I'd guess the thinking is that when liveness is enabled, there are multiple types of "results" (see one line below this, `selfie_liveness_results`).

Reply:
> * This spec should include a pair of context blocks to switch `liveness_checking_enabled`  on and off and check results

Implemented in fdcead1.

> * `doc_auth_result` existed previously, but maybe it should be normalized to `result` ? Or we include both? I'd guess the thinking is that when liveness is enabled, there are multiple types of "results" (see one line below this, `selfie_liveness_results`).

I'm leaning toward the "new" `doc_auth_result` being the correct value. At least, it matches what's expected based on CloudWatch logs, where `result` is not included in any log events.

```
filter name = 'IdV: doc auth image upload vendor submitted' and ispresent(properties.event_properties.result)
```

Reply:
Wow what a mess...didn't realize one suggestion would unearth so much cruft...

I guess this brings up a separate issue that our mocks do behave somewhat differently than the responses from our vendors, and IMO the whole point of the various response objects was that they were interchangeable value objects... so the subclasses shouldn't matter at all...

> I'm leaning toward the "new" doc_auth_result being the correct value. At least, it matches what's expected based on CloudWatch logs, where result is not included in any log events.

Yes I think we should have one value and always log that, it shouldn't be different based on which vendor we pick...

**separate follow-up idea** I were starting DocAuth from scratch, I'd let the things we have as response subclasses now be objects that process an HTTP response into a plain `FormResponse` or `DocAuth::Response` value object.... so that way all the responses are just the same class, as interchangeable as we can make them

Reply:
> **separate follow-up idea** I were starting DocAuth from scratch, I'd let the things we have as response subclasses now be objects that process an HTTP response into a plain `FormResponse` or `DocAuth::Response` value object.... so that way all the responses are just the same class, as interchangeable as we can make them

That makes sense to me. It would solve the current inconsistency between the constructor arguments, and it would force us to not rely on the availability of any methods or properties implemented only on the subclasses.

Reply:
Filed LG-4904 to make sure we don't lose that idea

Reply:
> Yes I think we should have one value and always log that, it shouldn't be different based on which vendor we pick...

dd4d519 pulls the mock response into alignment with Acuant and TrueID, so they should all be the same now among the doc auth responses.